libraries/AP_HAL_Linux/ConsoleDevice.cpp

```diff
@@ -0,0 +1,67 @@
+#include <AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+#include "ConsoleDevice.h"
+
+#include <stdio.h>
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+ConsoleDevice::ConsoleDevice() 
+{
+}
+
+ConsoleDevice::~ConsoleDevice()
+{
+
+}
+
+bool ConsoleDevice::close()
+{
+    if (::close(_rd_fd) < 0) {
```

Comment:
I don't think we want to close stdin/stdout
Instead just set a bool _closed flag and fail future operations